libraries/api-extractor-model/src/items/ApiDeclaredItem.ts

```diff
@@ -74,6 +79,20 @@ export class ApiDeclaredItem extends ApiDocumentedItem {
     return this._excerptTokens;
   }
 
+  /**
+   * The source file where the API item is declared. Is `undefined` if the
+   * source file could not be determined.
+   *
+   * @remarks
+   * The source file path is relative with respect to the `<projectFolder>` token as defined in
+   * the `api-extractor.json` config. It points to the `.ts` source file where the API item was
+   * originally declared. However, in some cases, if source map resolution fails, it falls back
+   * to pointing to the `.d.ts` file instead.
+   */
+  public get sourceFile(): string | undefined {
+    return this._sourceFile;
+  }
```

Comment:
[(issue comment r952034594)](https://github.com/microsoft/rushstack/pull/3590/files/5a4c0677a5c08bf190f54a7356a1d6179d6c3ac9#r952034594) raises a design question about how `projectFolderUrl` should be obtained.

For stereotypical Rush Stack projects, we always have a [repository field](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#repository) like this:

**package.json**
```
{
  "name": "@microsoft/api-extractor",
  "version": "7.29.2",
  "repository": {
    "type": "git",
    "url": "https://github.com/microsoft/rushstack.git", 👈👈👈
    "directory": "apps/api-extractor" 👈👈👈
  },
```

The ideal is to base it on these fields, however some nontrivial logic is required to combine these pieces to produce `https://github.com/microsoft/rushstack/tree/main/apps/api-extractor`. (And eventually we'd want to support other syntaxes such as GitLab, SVN, etc -- maybe someone already created an NPM package for this?)

For cases where the **package.json** data is missing or not correct, API Extractor should also provide a way to override this field explicitly.  I'm thinking that can be a setting such as `docModel.projectFolderUrl` in the [Doc Model Section](https://api-extractor.com/pages/configs/api-extractor_json/#doc-model-section) of **api-extractor.json**.

The **api-extractor.json** isn't the easiest to use, but it is easy to implement because it doesn't need any special URL parsing logic.  So if you're looking to save time, maybe that's a simpler option to start with? To demo this feature, we just need some way to obtain a working `projectFolderUrl` from the .api.json files -- we can always improve it later in another PR.

Reply:
I searched around a bit for an NPM package that can construct a "project folder URL" from the `package.json` `repository` field... but didn't have any luck. I went with the `api-extractor.json` based option instead, as you described.